cmd/integration/server_test.go

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2019 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/palantir/k8s-spark-scheduler-lib/pkg/apis/sparkscheduler/v1beta2"
+	ssclientset "github.com/palantir/k8s-spark-scheduler-lib/pkg/client/clientset/versioned/fake"
+	"github.com/palantir/k8s-spark-scheduler/cmd"
+	config2 "github.com/palantir/k8s-spark-scheduler/config"
+	"github.com/palantir/k8s-spark-scheduler/internal/binpacker"
+	"github.com/palantir/k8s-spark-scheduler/internal/common"
+	"github.com/palantir/witchcraft-go-server/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	apiextensionsfake "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	schedulerapi "k8s.io/kube-scheduler/extender/v1"
+)
+
+func Test_StaticCompaction(t *testing.T) {
+	rr := v1beta2.ResourceReservation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "appID1",
+			Namespace: "podNamespace",
+		},
+		Spec: v1beta2.ResourceReservationSpec{
+			Reservations: map[string]v1beta2.Reservation{
+				"driver": {
+					Node: "n1",
+					Resources: map[string]*resource.Quantity{
+						"cpu":    ToResource(resource.MustParse("1")),
+						"memory": ToResource(resource.MustParse("1024Mi")),
+					},
+				},
+				"executor-1": {
+					Node: "n1",
+					Resources: map[string]*resource.Quantity{
+						"cpu":    ToResource(resource.MustParse("2")),
+						"memory": ToResource(resource.MustParse("4096Mi")),
+					},
+				},
+				"executor-2": {
+					Node: "n1",
+					Resources: map[string]*resource.Quantity{
+						"cpu":    ToResource(resource.MustParse("2")),
+						"memory": ToResource(resource.MustParse("4096Mi")),
+					},
+				},
+			},
+		},
+		Status: v1beta2.ResourceReservationStatus{
+			Pods: map[string]string{
+				"driver":     "my-pod-driver",
+				"executor-1": "my-pod-executor-1",
+				"executor-2": "missing-pod-1",
```

Comment:
Are we aiming to test the case of a dead executors reservation being re-used by a new executor?

If we're just looking to test the happy case, we can remove this.

Reply:
No, this is testing the case where a pod used to exists, and then got re-scheduled